feat(font): add static optimization for fonts in expo router (#24027)
expo/expo: docs/pages/router/reference/static-rendering.mdx

}
```

## Fonts

> 

Author: ```suggestion

> Font optimization is available in SDK 50 and above.

```

> Expo Font has automatic static optimization for font loading in Expo Router. This means when you load a font with `expo-font`, Expo CLI will automatically extract the font resource and embed it in the page's HTML, enabling preloading, faster hydration, and reduced layout shift.
> 
> The following block will load Inter into the namespace and statically optimize on web:

Reviewer: ```suggestion
Expo Font has automatic static optimization for font loading in Expo Router. When you load a font with `expo-font`, Expo CLI will automatically extract the font resource and embed it in the page's HTML, enabling preloading, faster hydration, and reduced layout shift.

The following snippet will load Inter into the namespace and statically optimize on web:
```

We can be concise and also let's use snippet instead of block since we use that word elsewhere in the docs.

```js app/home.js
import { Text } from 'react-native';
import { useFonts } from 'expo-font';

export default function App() {
  /* @info Expo CLI automatically finds and extracts this font during compilation. */

>   const [isLoaded, error] = useFonts({

Reviewer: ```suggestion
  const [isLoaded] = useFonts({
```

Not sure but why are we referencing `error` here if we are not using it? Can `error` be used in the `if` condition on line 304? For example:

```
  if (!isLoaded && !error) {
    return null;
  }
```

    /* @end */
    inter: require('@/assets/inter.ttf'),
  });

  /* @info Always true on web with static rendering enabled. */
  if (!isLoaded) {
    /* @end */
    return null;
  }

  return <Text style={{ fontFamily: 'inter' }}>Hello Universe</Text>;
}
```

> This generates the following HTML in the static HTML:

Reviewer: ```suggestion
This generates the following snippet in the static HTML:
```

```html dist/home.html
/* @info preload the font before the JavaScript loads. */
<link rel="preload" href="/assets/inter.ttf" as="font" crossorigin />
/* @end */
<style id="expo-generated-fonts" type="text/css">
  @font-face {
    font-family: inter;
    src: url(/assets/inter.ttf);
    font-display: auto;
  }
</style>
```

- Static font optimization requires the font to be loaded syncronously. If the font isn't statically optimized, it could be because it was loaded inside a `useEffect`, deferred component, or async function.

> - Static optimization is only supported with `Font.loadAsync` and `Font.useFonts` from `expo-fonts`. Wrapper functions are supported, so long as the wrappers are synchronous.

Reviewer: ```suggestion
- Static optimization is only supported with `Font.loadAsync` and `Font.useFonts` from `expo-font`. Wrapper functions are supported as long as the wrappers are synchronous.
```

Typo in the library name.
